swarm/src/connection.rs

```diff
@@ -351,7 +355,16 @@ where
                         *shutdown = Shutdown::Later(Delay::new(dur), t)
                     }
                 }
-                (_, KeepAlive::No) => *shutdown = Shutdown::Asap,
+                (_, KeepAlive::No) => {
+                    // handle idle_timeout
+                    let duration = *idle_timeout; // Default timeout is 0 seconds
+                    if duration > Duration::new(0, 0) {
```

Comment:
I think there is a constant for 0.

Reply:
https://doc.rust-lang.org/std/time/struct.Duration.html#associatedconstant.ZERO